IIPS/iip-X.md

```diff
@@ -0,0 +1,137 @@
+
+It's common nowadays that using URI with certain convention mapping to the specific resources. With this experience, we utilize the advantage of blockchain to provide blockchain domain names as short, stable, and human-readable identifiers used to specify network resources. In this way, users can enter a memorable string, such as `loopchain.icon`, just like how we normally do on browser, and be directed to the appropriate resource. The mapping between names and resources may change in any scenarios, so a website may change the corresponding IPFS hashes, projects may update their smart contract, a user may change wallets, or a IPFS document may be updated to a new version, without the domain name changing. Further, a domain need not specify a single resource; different record types allow the same domain to reference different resources. For instance, the browser may resolve `loopchain.icon` to the IPFS hash to redirect to IPFS content, even resolve the same address to a smart contract.
+
+## Motivation
+
+Hash is the fundamental of blockchain and used to represent an account of an identity, but the hash address is not easily remember for the normal user, by integrate the ICON Name Service will bring three following advantages:
+
+- Human-readable identity, users can easily catch up with the name like `loopchain.icon` instead of hash address.
+- Multiple services under a single name, such as a DApp hosted in IPFS, a contract address, and even a mail server.
+- Increaing the security by tamper proof of the accessing smart contract for the mapping address.
+
+These three features will lower the gap of blockchain technologies and improvement the user adoption.
+
+## Specification
+
+The INS system has thsee main parts:
+
+### Registry
+INS registry is a single contract that provides a mapping from any registered name to the resolver responsible for it, and permits the owner of a name to set the resolver address, and to create subdomains, potentially with different owners to the parent domain.
+
+### Registrars
+INS registrar is a smart contract that owns a Top-level domain, and allocates subdomains of it according to some set of rules defined in the smart contract code. For any available domain name, users will be able to purchase. The rules for the INS registrar smart contract will be interatly updated according to meet the benefit of whole ICON community.
+
+### Resolvers
+INS resolvers are responsible for performing resources from a domain name, a resources can be a contract address, wallet address and IPFS hash. Utilizing the resolver smart contract enables users and developers using the decentralized resources easily bring more ideas and values to the ecosystem.
+
+### Reverse Resolvers
+INS reverse resolver is a smart contract records that mapping from addresses to names.
+
+## Rationale
```

Comment:
Is this right header name? It looks the following part actually explains real specification, while the above part explains overview. I think this 'Rationale' header should be removed.